src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -62,6 +62,32 @@ public partial class ItemsControl : Control, IItemsControl
 
 		internal ScrollViewer ScrollViewer { get; private set; }
 
+		/// <summary>
+		/// Stores materialized containers starting a given index, so that their
+		/// ItemsControl.IndexForContainerProperty can be updated after the collection changes.		
+		/// </summary>
+		/// <param name="startingIndex">The minimum index of containers we care about.</param>
+		/// <param name="indexChange">How does the index change.</param>
+		private void SaveContainersAndRepairIndex(int startingIndex, int indexChange)
+		{
+			Dictionary<DependencyObject, object> _containers = new Dictionary<DependencyObject, object>();
```

Comment:
why create this dictionary at all any more? Just set IndexForItemContainerProperty directly from the first foreach loop, you don't need two loops.

Reply:
I forgot to review the summary